app/src/main/java/com/example/taskyy/ui/viewmodels/AgendaViewModel.kt

```diff
@@ -25,21 +26,62 @@ class AgendaViewModel @Inject constructor(
 
     fun onEvent(event: AgendaEvent) {
         when (event) {
-            is AgendaEvent.OnMonthExpanded -> state = state.copy(isMonthExpanded = event.isMonthExpanded)
+            is AgendaEvent.OnMonthExpanded -> state =
+                state.copy(isMonthExpanded = event.isMonthExpanded)
+
             is AgendaEvent.OnDateSelected -> formatSelectedDate(event.date)
-            is AgendaEvent.OnUserInitialsClicked -> state = state.copy(isUserDropDownExpanded = event.isUserDropDownExpanded)
+            is AgendaEvent.OnUserInitialsClicked -> state =
+                state.copy(isUserDropDownExpanded = event.isUserDropDownExpanded)
+
             is AgendaEvent.OnLogOutCLicked -> logout()
+            is AgendaEvent.AddAgendaItem -> state =
+                state.copy(isAddAgendaItemExpanded = event.isAgendaItemExpanded)
+
+            is AgendaEvent.OnAgendaDaySelected -> state =
+                state.copy(selectedAgendaDay = event.agendaDaySelected)
         }
     }
 
     private fun formatSelectedDate(date: Long) {
+
         val instant = Instant.ofEpochMilli(date)
         val localDateTime = LocalDateTime.ofInstant(
             instant,
             ZoneId.systemDefault()
-        ) // or specify your desired time zone
-        val formatter = DateTimeFormatter.ofPattern("MMMM", Locale.ENGLISH)
-        state = state.copy(selectedMonth = formatter.format(localDateTime).uppercase())
+        )
+
+        val monthFormatter = DateTimeFormatter.ofPattern("MMMM", Locale.ENGLISH)
+        val dayOfTheMonthFormatter = DateTimeFormatter.ofPattern("d", Locale.ENGLISH)
+        val dayOfTheWeekFormatter = DateTimeFormatter.ofPattern("E", Locale.ENGLISH)
+
+        val tempDateList: ArrayList<LocalDateTime> = java.util.ArrayList()
+        val dayList: ArrayList<Day> = ArrayList()
+
+        val dateList: ArrayList<LocalDateTime> = incrementDate(localDateTime, tempDateList)
+
+        for (date in dateList) {
+            val day = Day(
+                dayOfTheWeek = dayOfTheWeekFormatter.format(date),
+                dayOfTheMonth = dayOfTheMonthFormatter.format(date)
+            )
+            dayList.add(day)
+        }
+
+
+        state = state.copy(
+            selectedMonth = monthFormatter.format(localDateTime).uppercase(),
+            selectedDay = dayList,
+        )
+    }
```

Comment:
I'm very confused about what this function does 😅  Why do you need 3 ArrayLists? 
Apparently the only goal is to format the month and days list. You first construct a temp date list and a day list, just to then construct a new temporary dateList which gets then assigned to the dayList. Why not populate the dayList right away?
To construct the complete day list, all you need is this:
```
val days = (0..6).map {
    val date = localDateTime.plusDays(i.toLong()) 
    Day(
        dayOfTheWeek = dayOfTheWeekFormatter.format(date),
        dayOfTheMonth = dayOfTheMonthFormatter.format(date)
    )
}
```